vunit/ui.py

```diff
@@ -1032,9 +1032,21 @@ def _compile(self, simulator_if):
         """
         Compile entire project
         """
+        # get test benches
+        if self._args.minimal:
+            target_files = self._get_testbench_files(simulator_if)
+        else:
+            target_files = None
+
         simulator_if.compile_project(self._project,
                                      continue_on_error=self._args.keep_compiling,
-                                     printer=self._printer)
+                                     printer=self._printer, target_files=target_files)
+
+    def _get_testbench_files(self, simulator_if):
```

Comment:
I think this function misses a docstring explaning what it does. Such as "Return the list of all files for the currently selected tests to run"